lib/libzpool/uzfs_io.c

```diff
@@ -99,15 +99,12 @@ uzfs_write_data(zvol_state_t *zv, char *buf, uint64_t offset, uint64_t len,
 
 	rl = zfs_range_lock(&zv->zv_range_lock, offset, len, RL_WRITER);
 
-	if (!is_rebuild && (zv->zv_status & ZVOL_STATUS_DEGRADED))
-		uzfs_add_to_incoming_io_tree(zv, offset, len);
-
-	if (zv->zv_rebuild_status & ZVOL_REBUILDING_IN_PROGRESS) {
-		if (is_rebuild) {
-			count = uzfs_search_incoming_io_tree(zv, offset,
-			    len, (void **)&chunk_io);
-			if (!count)
-				goto exit_with_error;
+	if ((zv->zv_status & ZVOL_REBUILDING_IN_PROGRESS) &&
```

Comment:
zv_status is a variable to maintain state not flags. For improving readability better to use  "=="/"logical equal" to operator.

Reply:
ok